### DISCUSSION TITLE
TinyUSB (CDC+MSC) Keil STM32F0 Not Working

### DISCUSSION THREAD

**Discussion by g-Mutlu** [hathach/tinyusb#568] [General]
Hi Guys,

Used the example in the tinyusb folder for MSC. (tinyusb-master\examples\device\cdc_msc\src)
I can successfully run the same code in STM32CubeIDE. However, Keil didn't happen in Uvision5. Does not recognize the USB device. Can you help me?
Note: In STM32CubeIDE, the compiler uses __GNUC__.
I had to enable -gnu to compile it in keil.

![image](https://user-images.githubusercontent.com/64061217/102478083-3aae1700-406e-11eb-855e-ba5e406be330.png)
Code:
[STM32F0_MASS_STORAGE_V00.zip](https://github.com/hathach/tinyusb/files/5708752/STM32F0_MASS_STORAGE_V00.zip)

**Comment by hathach** [comment]
have you tried to compile it with current make file. Maybe you miss some of macros, config etc with the cube IDE

**Reply by AshishvSpanidea** [reply]
> Hi Guys,
> 
> Used the example in the tinyusb folder for MSC. (tinyusb-master\examples\device\cdc_msc\src) I can successfully run the same code in STM32CubeIDE. However, Keil didn't happen in Uvision5. Does not recognize the USB device. Can you help me? Note: In STM32CubeIDE, the compiler uses **GNUC**. I had to enable -gnu to compile it in keil.
> 
> ![image](https://user-images.githubusercontent.com/64061217/102478083-3aae1700-406e-11eb-855e-ba5e406be330.png) Code: [STM32F0_MASS_STORAGE_V00.zip](https://github.com/hathach/tinyusb/files/5708752/STM32F0_MASS_STORAGE_V00.zip)

Hey @g-Mutlu what was the STM32 board you used. I have STM32F072 Discovery Board. I had followed both #633 and https://anvilelectronics.ovh/tinyusb-na-stm32-device-cdc/ steps to implement tinyUSB in STM32CubeIDE but it couldn't enumerate.
Any suggestions.

Regards
Ashish
